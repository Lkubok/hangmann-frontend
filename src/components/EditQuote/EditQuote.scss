@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";

.edit-wrapper {
  @include flex-cc-row;
  width: 100%;
}
