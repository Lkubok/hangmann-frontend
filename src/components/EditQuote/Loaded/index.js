import Loaded from "./Loaded";
export default Loaded;
