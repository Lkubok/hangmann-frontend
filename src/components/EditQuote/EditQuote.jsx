import React, { PureComponent } from "react";
import { connect } from "react-redux";
import { fetchQuote } from "../../actions/editQuoteActions";
import * as selectors from "../../reducers/selectors";
import Loading from "./Loading";
import Loaded from "./Loaded";
import "./EditQuote.scss";

export class EditQuote extends PureComponent {
  componentDidMount() {
    const { quoteId } = this.props.match.params;
    this.props.fetchQuote(quoteId);
  }
  renderContent = () => {
    return this.props.isFetching ? (
      <Loading />
    ) : (
      <Loaded quoteId={this.props.match.params.quoteId} />
    );
  };
  render() {
    return <div className="edit-wrapper">{this.renderContent()}</div>;
  }
}

const mapStateToProps = state => ({
  isFetching: selectors.getFetchingStatus(state)
});

const mapDispatchToProps = {
  fetchQuote
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditQuote);
