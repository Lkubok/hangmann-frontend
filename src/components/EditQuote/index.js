import EditQuote from "./EditQuote";
export default EditQuote;
