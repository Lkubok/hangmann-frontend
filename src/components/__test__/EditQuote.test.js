import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { EditQuote } from "../EditQuote/EditQuote";
Enzyme.configure({ adapter: new Adapter() });

describe("EditQuote component", () => {
  let props;
  const spyFunction = jest.fn();
  beforeEach(() => {
    props = {
      match: {
        params: {
          quoteId: "123"
        }
      },
      fetchQuote: spyFunction
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<EditQuote {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
