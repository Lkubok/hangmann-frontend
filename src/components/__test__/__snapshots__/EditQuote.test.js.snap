// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditQuote component Should match snapshot 1`] = `
<div
  className="edit-wrapper"
>
  <Connect(Loaded)
    quoteId="123"
  />
</div>
`;
